### DISCUSSION TITLE
CMake Errors / Win 10 / VS 2022

### DISCUSSION THREAD

**Discussion by Kybernaut000** [alandefreitas/matplotplusplus#410] [Q&A] [answered]
Hi,

I'm trying to get matplot++ compiled in Win 10. I found an older question from 2022, but it is unanswered.

I tried to follow your guide on building from source. I did the following:
- Installed GnuPlot (latest version)
- Installed git
- Installed Visual Studio 2022
-  Cloned the repository within Visual Studio 2022

I directly get the following CMake errors:

CMake was unable to find a build program corresponding to "Unix Makefiles".
CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.

CMAKE_C_COMPILER not set, after EnableLanguage

CMAKE_CXX_COMPILER not set, after EnableLanguage	

(I get the same errors when cloning with external/local git instead of VS 2022's git.)

I found the CMakePresets_schema.json, and I guess I have to define my own configurePreset and buildPreset. But I'm completely unfamiliar with CMake. I also tried adding the path of the MS Visual Studio 2022 compiler to the CMakeList.txt, but I guess this will not  be sufficient, and I'm not sure what to add ("cl.exe"?).

Can you please hint me in any direction? If matplot++ is not supported with Windows (any more), please let me know.

Regards,
Alex

**Comment by alandefreitas** [comment]
> CMake was unable to find a build program corresponding to "Unix Makefiles".

You shouldn't set the generator or compilers when using MSVC.

**Reply by Kybernaut000** [reply]
Hi Alan,

thank you for your ultra-quick response. I didn't set or change the generator.

Maybe I expressed myself ambiguously:

- I installed VS 2022 (no modifications, Community Edition)
- I cloned the project via git inside VS 2022
- I didn't change any file

The error message comes directly after cloning, from the cmake-toolchain installed automatically with VS 2022.

So, I don't know what to.do from here on.

Greetings,
Alex

**Comment by Kybernaut000** [comment, marked as answer]
Hello,

I solved it. Might be completely obvious for anyone who's into CMake, but for me it took a while. As expected the problem was in CmakePresets.json. The generator is "Unix makefiles". I had to change this to "Visual Studio 17 2022" and was able to make and compile it.

So basically I did:

```
"configurePresets": [
     {
-      "name": "dev",
-      "displayName": "Development Mode",
+      "name": "devUNIX",
+      "displayName": "Development Mode - Unix Makefiles",
       "description": "Build with warnings and debug symbols",
       "binaryDir": "build",
       "generator": "Unix Makefiles",
```

and

```
+    {
+      "name": "devMSVC2022",
+      "displayName": "Development Mode - Visual Studio 2022",
+      "description": "Build with warnings and debug symbols",
+      "binaryDir": "build",
+      "generator": "Visual Studio 17 2022",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+      }
+    },
```

and

```
   "buildPresets": [
     {
-      "name": "dev",
-      "configurePreset": "dev",
+      "name": "devUNIX",
+      "configurePreset": "devUNIX",
       "configuration": "RelWithDebInfo",
       "jobs": 8
     },
```

and so on.

As I don't have write privilege, I cannot push my committ. Please find it attached. Let me know, if this is fine for you.

Thanks again,
Alex

[0001-build-Added-support-for-MS-Visual-Studio-2022-in-CMa.patch](https://github.com/user-attachments/files/16073875/0001-build-Added-support-for-MS-Visual-Studio-2022-in-CMa.patch)

**Comment by alandefreitas** [comment]
Yes. That's correct. CMakePresets.cmake should't explicitly set the generator.

**Reply by Kybernaut000** [reply]
Unfortunately, the CmakePresets.json in the repository does just that. Inthink it would be a good idea to change that, but of course this is up to you.

**Reply by alandefreitas** [reply]
Yes. It's a contribution we accepted but it shouldn't have the `"generator": "Unix Makefiles",` entry.
